### DISCUSSION TITLE
New deploy of Next.js app to Vercel - 502: Bad_Gateway Error

### DISCUSSION THREAD

**Discussion by taylorhlabs** [prisma/prisma#4848] [Q&A] [answered]
I just deployed a new Next app (with Prisma v2.13.1) to Vercel and I'm getting the following error in the functions log:

```
ERROR
Unhandled error during request: PrismaClientInitializationError3 [PrismaClientInitializationError]: error:
Error validating datasource `db`: The URL for datasource `db` must start with the protocol `mysql://`.
  -->  schema.prisma:7
   | 
 6 |   provider = "mysql"
 7 |   url      = env("DATABASE_URL")
   | 
Validation Error Count: 1
    at PrismaClientFetcher.request (/var/task/node_modules/@prisma/client/runtime/index.js:78127:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  clientVersion: '2.13.1'
}
```

I followed the [Vercel Deployment guide](https://www.prisma.io/docs/guides/deployment/deploying-to-vercel) for adding database environmental variables (since .env file is only used in local environment) and I included the vercel.json file (below) as instructed in deployment guide.

```
{
  "version": 2,
  "env": {
    "DATABASE_URL": "@database_url"
  },
  "build": {
    "env": {
      "DATABASE_URL": "@database_url"
    }
  }
}
```
The app is working fine with a hosted MySQL database in local environment. Any idea why it wouldn't be working in production?

**Comment by taylorhlabs** [comment, marked as answer]
I resolved the error by removing the vercel.json file from my project. I believe the Prisma Vercel Deployment Guide should be updated as the vercel.json file is no longer required for defining Environment Variables per this [Vercel blog post](https://vercel.com/blog/environment-variables-ui). You can now define Env Vars directly in the Vercel Dashboard when deploying a project. (Note I wasn't using the Vercel CLI to deploy my Next.js Prisma Project as recommended in the Prisma Vercel Deployment Guide. It's a breeze using the Vercel Dashboard...)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
